test/test.mjs

```diff
@@ -1078,25 +1078,33 @@ async function main() {
     stats = [];
   }
 
-  if (options.downloadOnly) {
-    await ensurePDFsDownloaded();
-  } else if (options.unitTest) {
-    // Allows linked PDF files in unit-tests as well.
-    await ensurePDFsDownloaded();
-    startUnitTest("/test/unit/unit_test.html", "unit");
-  } else if (options.fontTest) {
-    startUnitTest("/test/font/font_test.html", "font");
-  } else if (options.integration) {
-    // Allows linked PDF files in integration-tests as well.
-    await ensurePDFsDownloaded();
-    startIntegrationTest();
-  } else {
-    startRefTest(options.masterMode, options.reftest);
+  try {
+    if (options.downloadOnly) {
+      await ensurePDFsDownloaded();
+    } else if (options.unitTest) {
+      // Allows linked PDF files in unit-tests as well.
+      await ensurePDFsDownloaded();
+      await startUnitTest("/test/unit/unit_test.html", "unit");
+    } else if (options.fontTest) {
+      await startUnitTest("/test/font/font_test.html", "font");
+    } else if (options.integration) {
+      // Allows linked PDF files in integration-tests as well.
+      await ensurePDFsDownloaded();
+      await startIntegrationTest();
+    } else {
+      await startRefTest(options.masterMode, options.reftest);
+    }
+  } catch (e) {
+    // Close the browsers if uncaught exceptions occur, otherwise the spawned
+    // processes can become orphaned and keep running after `test.mjs` exits
+    // because the teardown logic of the tests did not get a chance to run.
+    console.error(e);
+    await Promise.all(sessions.map(session => closeSession(session.name)));
   }
 }
 
 var server;
-var sessions;
+var sessions = [];
```

Comment:
Given this change, can the following line be removed now?
https://github.com/mozilla/pdf.js/blob/db9115625b9e16b79ffdfcd8481547ae6f05a90a/test/test.mjs#L974

Reply:
Good catch; yes, that's possible now. I have removed it in the new commit.